functions/src/definitions/userHandlers.js

```diff
@@ -275,96 +287,194 @@ async function newImageInstanceHandler({
 }) {
   const db = admin.firestore()
   let filename
+  let messageRef
+  let hasMatch = false
+  let captionHash = caption ? hashMessage(caption) : null
+
   //get response buffer
   let buffer = await downloadWhatsappMedia(mediaId, mimeType)
   const hash = await getHash(buffer)
+
+  //upload to storage
+  const storageBucket = admin.storage().bucket()
+  filename = `images/${mediaId}.${mimeType.split("/")[1]}`
+  const file = storageBucket.file(filename)
+  const stream = file.createWriteStream()
+  await new Promise((resolve, reject) => {
```

Comment:
Hash can be checked before uploading to avoid having the storage grow

Reply:
Done